From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Simon Gardling <titaniumtown@gmail.com>
Date: Wed, 10 Nov 2021 12:38:09 -0500
Subject: [PATCH] bump dependencies


diff --git a/build.gradle.kts b/build.gradle.kts
index 92b612126a6bac0b89198a92bbb73b742ec9d064..0cb3697fe3755d367f48c4172be31bdf7c931e2c 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -26,12 +26,12 @@ configurations.api {
 dependencies {
     // api dependencies are listed transitively to API consumers
     api("commons-lang:commons-lang:2.6")
-    api("com.google.guava:guava:21.0")
-    api("com.google.code.gson:gson:2.8.8") // Paper - bump
+    api("com.google.guava:guava:31.0.1-jre")
+    api("com.google.code.gson:gson:2.8.9") // Paper - bump
     api("net.md-5:bungeecord-chat:1.16-R0.4")
     api("org.yaml:snakeyaml:1.28")
     api("com.googlecode.json-simple:json-simple:1.1.1") // Paper
-    api("it.unimi.dsi:fastutil:8.2.2")
+    api("it.unimi.dsi:fastutil:8.5.6")
     apiAndDocs(platform("net.kyori:adventure-bom:$adventureVersion"))
     apiAndDocs("net.kyori:adventure-api")
     apiAndDocs("net.kyori:adventure-text-serializer-gson")
@@ -40,8 +40,8 @@ dependencies {
     api("org.apache.logging.log4j:log4j-api:2.14.1") // Paper
     api("org.slf4j:slf4j-api:1.7.30") // Paper
 
-    implementation("org.ow2.asm:asm:9.1")
-    implementation("org.ow2.asm:asm-commons:9.1")
+    implementation("org.ow2.asm:asm:9.2")
+    implementation("org.ow2.asm:asm-commons:9.2")
 
     compileOnly("org.apache.maven:maven-resolver-provider:3.8.1")
     compileOnly("org.apache.maven.resolver:maven-resolver-connector-basic:1.7.0")
