From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Simon Gardling <titaniumtown@gmail.com>
Date: Mon, 8 Nov 2021 22:28:08 -0500
Subject: [PATCH] update dependencies


diff --git a/build.gradle.kts b/build.gradle.kts
index 36e731f3f05648a80ccba96a8193bab1b3766240..b305cb35100b0cee6ccb0740ea3ad18179540f31 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -52,7 +52,7 @@ dependencies {
     runtimeOnly("org.apache.maven.resolver:maven-resolver-connector-basic:1.7.0")
     runtimeOnly("org.apache.maven.resolver:maven-resolver-transport-http:1.7.0")
 
-    implementation("io.netty:netty-all:4.1.65.Final") // Paper
+    implementation("io.netty:netty-all:4.1.70.Final") // Paper
 
     implementation("net.fabricmc:mapping-io:0.3.0") // Paper - needed to read mappings for stacktrace deobfuscation
     implementation("com.velocitypowered:velocity-native:1.1.0-SNAPSHOT") // Paper
