From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KurumiFake <kurumiisshidowife@gmail.com>
Date: Thu, 3 Feb 2022 19:46:17 +0700
Subject: [PATCH] a


diff --git a/src/main/java/net/minecraft/core/Direction.java b/src/main/java/net/minecraft/core/Direction.java
index 268b631716973d24607d3c4763ac8cdb1ebc4a9b..8e8ecedc5571fdd11e171f80758e423090d8f3fd 100644
--- a/src/main/java/net/minecraft/core/Direction.java
+++ b/src/main/java/net/minecraft/core/Direction.java
@@ -68,6 +68,8 @@ public enum Direction implements StringRepresentable {
     private final int adjZ;
     // Paper end
 
+    public static final Direction[] values = values();
+
     private Direction(int id, int idOpposite, int idHorizontal, String name, Direction.AxisDirection direction, Direction.Axis axis, Vec3i vector) {
         this.data3d = id;
         this.data2d = idHorizontal;
