From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joseph Burton <burtonjae@hotmail.co.uk>
Date: Mon, 25 Apr 2022 18:06:04 +0200
Subject: [PATCH] (Paper-1.18.2) Fix V1451 dataconverter stat types


diff --git a/src/main/java/ca/spottedleaf/dataconverter/minecraft/versions/V1451.java b/src/main/java/ca/spottedleaf/dataconverter/minecraft/versions/V1451.java
index a20d258814b0d2d0fa01d45be43a66987de19598..a1f270f55617e1608bf7821951fa589e611b6f7d 100644
--- a/src/main/java/ca/spottedleaf/dataconverter/minecraft/versions/V1451.java
+++ b/src/main/java/ca/spottedleaf/dataconverter/minecraft/versions/V1451.java
@@ -391,7 +391,7 @@ public final class V1451 {
 
             switch (type) {
                 case "minecraft:mined": {
-                    WalkerUtils.convert(MCTypeRegistry.ITEM_NAME, criteriaType, "id", fromVersion, toVersion);
+                    WalkerUtils.convert(MCTypeRegistry.BLOCK_NAME, criteriaType, "id", fromVersion, toVersion);
                     break;
                 }
 
@@ -400,7 +400,7 @@ public final class V1451 {
                 case "minecraft:broken":
                 case "minecraft:picked_up":
                 case "minecraft:dropped": {
-                    WalkerUtils.convert(MCTypeRegistry.BLOCK_NAME, criteriaType, "id", fromVersion, toVersion);
+                    WalkerUtils.convert(MCTypeRegistry.ITEM_NAME, criteriaType, "id", fromVersion, toVersion);
                     break;
                 }
 
